core/src/apps/cardano/auxiliary_data.py

```diff
@@ -142,19 +160,39 @@ async def _show_governance_registration(
             ctx, encoded_public_key, delegation.weight
         )
 
+    # For non-payment governance reward addresses, we show a warning that the address is not
+    # actually eligible for rewards. https://github.com/cardano-foundation/CIPs/pull/373
+    # However, the registration is otherwise valid, so we allow such addresses since we don't
+    # want to prevent the user from voting just because they use an outdated wallet.
+    if parameters.reward_address:
+        show_payment_warning = (
+            addresses.get_type(addresses.get_bytes_unsafe(parameters.reward_address))
+            not in addresses.ADDRESS_TYPES_PAYMENT
+        )
```

Comment:
nit: IMO it's worth it to extract this into a function taking just `address_type` as parameter and then reusing it also with address params.